src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/TerminateDebugHandler.ts

```diff
@@ -12,27 +12,47 @@ export async function onDidTerminateDebugSession(
   event: DebugSession,
   logger: RazorLogger,
   targetPid: number | undefined,
+  targetProcess: string | boolean = false,
 ) {
-  if (event.type !== 'pwa-chrome' && event.type !== 'pwa-msedge') {
+  // Ignore debug sessions that are not applicable to us
+  const VALID_EVENT_TYPES = ['pwa-chrome', 'pwa-msedge', 'coreclr'];
```

Comment:
Totally ignorant question but if a user launches a classic web ap side-by-side a hosted Blazor WASM application will this terminate session do the right thing and only terminate the hosted Blazor wasm app?

Reply:
Not ignorant at all. In fact, quite astute.

In the scenario you described, the Blazor WASM session would be terminated as well.

To remedy this, I used the event name as a unique identifier for the debug session so we match event names based on what we call them.

I do notice that DebugSessions have a unique identifier associated with them. I'll have to see if it is possible to see if there is a way to return this unique identifier from the `startDebugging` API.